design/ceph/object/swift-and-keystone-integration.md

```diff
@@ -0,0 +1,312 @@
+  present.
+
+The certificate to verify the Keystone endpoint can't be explicitly
+configured in Ceph RGW, the system configuration of the pod is used,
+you can add to the system certificate store via the
+`gateway.caBundleRef` setting of the object store CRD.
+
+The credentials for the Keystone service account used by Ceph RGW are
+supplied in a Secret that contains a mapping of OpenStack [openrc
+environment variables](https://docs.openstack.org/python-openstackclient/xena/cli/man/openstack.html#environment-variables).
+`password` is the only authentication type that is supported, this is
+a limitation of RGW which does not support other Keystone authentication
+types. Example:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: rgw-service-user
+data:
+  OS_PASSWORD: "horse staples battery correct"
+  OS_USERNAME: "ceph-rgw"
+  OS_PROJECT_NAME: "admin"
+  OS_USER_DOMAIN_NAME: "Default"
+  OS_PROJECT_DOMAIN_NAME: "Default"
+  OS_AUTH_TYPE: "password"
+```
+
+This format is chosen because it is a natural and interoperable way
+that keystone credentials are represented and for the supported auth
+type it maps naturally to the Ceph RGW configuration.
+
+The following constraints must be fulfilled by the secret:
+* `OS_AUTH_TYPE` must be `password` or omitted.
+* `OS_USER_DOMAIN_NAME` must equal `OS_PROJECT_DOMAIN_NAME`. This is a
+  restriction of Ceph RGW, which does not support configuring separate
+  domains for the user and project.
+* All openrc variables not in the example above are ignored. The API
+  version (`OS_IDENTITY_API_VERSION`) is assumed to be `3` and
+  Keystone endpoint `OS_AUTH_URL` is taken from the
+  `keystone.apiVersion` configuration in the object store resource.
+
+The mapping to
+[RGW configuration options](https://docs.ceph.com/en/octopus/radosgw/config-ref/#keystone-settings)
+is done as follows:
+* `OS_USERNAME` -> `rgw_keystone_admin_user`
+* `OS_PROJECT_NAME` -> `rgw_keystone_admin_project`
+* `OS_PROJECT_DOMAIN_NAME`, `OS_USER_DOMAIN_NAME` -> `rgw_keystone_admin_domain`
+* `OS_PASSWORD` -> `rgw_keystone_admin_password`
+
+### Swift integration
+
+The currently ignored `gateway.type` option is deprecated and from now on
+explicitly ignored by rook.
+
+The other `gateway` settings are kept as they are: They do not directly
+relate to Swift or S3 but are common configuration of RGW.
+
+The Swift API is enabled and configured via a new `protocols` section:
+```yaml
+protocols:
+  swift:                      [1]
+    accountInUrl: true        [*]
+    urlPrefix: /example       [*]
+    versioningEnabled: false  [*]
+  s3:
+    enabled: false            [2]
+```
+Annotations:
+* `[1]` Swift will be enabled, if `protocols.swift` is present.
+* `[2]` This defaults to `true` (even if `protocols.s3` is not present
+  in the CRD). This maintains backwards compatibility – by default S3
+  is enabled.
+* `[*]` These options map directly to [RGW configuration
```

Comment:
nit: Why not name it numerically? I'm confused at a glance why the asterisk is used.
```suggestion
* `[3]` These options map directly to [RGW configuration
```

Reply:
I felt using symbols for annotations used several times may help with readability. The numerical, single use annotations can then be used in order in the code example (making the lookup easy), while the symbol stands out from the numbers, which helps me with the "mental lookup". If you prefer numerical reference labels I have no real objection.

Reply:
Ok, not a big deal really, just wanted to understand. Personally I would suggest to keep all numbers or all symbols, but I'm open to new styles too. :)

Reply:
I think remaining consistent with the rest of the documents is reason enough to make them all numerical. Especially, since the mixed style already confused at least one person. I've changed them to all numbers now.